### ISSUE TITLE
[FEATURE REQUEST] A faster turnaround time for updates! Or you'll lose users. You already are, though.

### ISSUE THREAD

**Issue by bytorphoto** [valheimPlus/ValheimPlus#807]
It seems to me that the turnaround time for fixes to V+ are glacially slow. Something needs to be implemented to improve that speed, or users will leave in their droves.

This is not a bitching session, I've loved using V+, but the last couple of updates, including the current one, are making me wonder if V+ is worth waiting for. The product is ubiquitous, and the slow refresh rate has made a LOT of private servers unplayable now. Or server admins are just turning it off and going other routes.

PLEASE set up something that will allow updates to turnaround faster. This latest issue is already 3 weeks old and more than 3 Valheim bugfixes past.

**Comment by SSimp258** [comment]
There are things that happen IRL that are beyond people's control. I would advise you on them but I'm not at liberty to explain what's going on. I would also advise you that there is a community fix while things are going on that people are using in the discord and other various locations. You, and these servers, should be utilizing it.

You can close this issue as it's a non-issue.

**Comment by bytorphoto** [comment]
It's honestly not a non-issue as many dedicated server providers will only install "stable" release versions, not community updates, thus rendering them unusable. But I'll close it.

And I know that IRL stuff happens. I've been working with computers for over 40 years, both in support and development, and also realise that others need to be able to commit changes, fixes and updates, rather than just one or two people that aren't available all the time. We shouldn't have to rely on "community patches".

But as you say... it's a "non-issue".

**Comment by SSimp258** [comment]
You're right. We shouldn't have to rely on the community to keep the mod going but we can't change that now. NX said he was going to try to prevent this but things happened and he never got around to it. Most hosted servers that we deal with on the discord have switched over to the community fix. I would caution against using other community fixes as they may not have their source code on GIT for viewing so they can add in code to infect your computer.

**Comment by bluesort** [comment]
Is the community fix documented anywhere outside the Discord? It'd ideally be somewhere searchable

**Comment by Dicebar** [comment]
I'm not so much curious about what happened, but more about what _will_ happen. Is the problem _for the addon's updates_ simply an AWOL NX? Will you move repos, rename, and carry on? Will you wait and pray to your deity of choice until NX returns? What is the plan for the future?

**Comment by SSimp258** [comment]
> Is the community fix documented anywhere outside the Discord? It'd ideally be somewhere searchable

I'm not sure if this will answer your question but if you go to Issue #793 that's where we troubleshot what broke when Valheim updated. We point ppl in discord to that issue to help them figure out their servers/clients. There's a lot of comments from others on how they got their stuff to work on there as well as links from Grantapher repo for files.

@Dicebar Your questions are a lot harder to answer. I can't say exactly. There are times that I think about leaving the discord because of all the vitrol from others when the mod isn't updated within their expectations. They want V+ updated within seconds of Valheim being updated which will never happen. I've stepped up kicking people from the discord that are either very aggressive or even harshly passive-aggressive with their comments because the vitrol isn't needed. You've probably seen my memes when ppl complain so i'm hopeful that making memes about ppl complaining will get them to layoff the devs. I see my "job" as a chat mod to make sure the vitrol stops with me so the devs don't see it basically.

**Comment by Grantapher** [comment]
@CarterJackson Since we can't change this repo there is only so many ways to point to https://github.com/Grantapher/ValheimPlus/releases/latest. It is listed on thunderstore and nexus as well.

**Comment by bluesort** [comment]
@Grantapher Nice. I'm using the Docker container, so I'll have to apply the fix differently, but I saw a method mentioned in the thread @SSimp258 linked so I'll take a stab at it when I have time. Thank you both

**Comment by Grantapher** [comment]
Which Docker container?

**Comment by bluesort** [comment]
@Grantapher [lloesche's](https://github.com/lloesche/valheim-server-docker), which to my knowledge is the most common one. Mentioned in the other thread [here](https://github.com/valheimPlus/ValheimPlus/pull/793#issuecomment-1475127777) along with some follow-up context, but I've not tried it myself yet

**Comment by Grantapher** [comment]
I also use that one, and it is even easier now since lloesche/valheim-server-docker#599

Add an env var: 

```
+  VALHEIM_PLUS_REPO: "Grantapher/ValheimPlus",
```

[You can find that env var doc here](https://github.com/lloesche/valheim-server-docker#:~:text=with%20BEPINEX.-,VALHEIM_PLUS_REPO,-valheimPlus/ValheimPlus)

**Comment by Dicebar** [comment]
@SSimp258 I'm sorry to hear you guys are receiving so much vitriol. I'm an addon maintainer myself, so I know how RL can get in the way and how much time and effort it costs. I've found that expectation management helps a lot, though. For users (or anyone, really) waiting for "eventually, maybe" is very frustrating, so I would like to encourage you all making some sort of plan you can share with your users.

**Comment by bluesort** [comment]
@Grantapher That's awesome, thanks! You and lloesche rock.

**Comment by SSimp258** [comment]
@Dicebar So the plan was for NX to add some maintainers to the GIT repo that could publish and approve PRs but he was pulled away like right after that so the "plan' so to speak, was never put into place and he's pretty hard to get in touch with from everything going on. So we did have a plan but was just never put into motion.

**Comment by vividnightmare** [comment]
Forgive me if I missed something, but TL;DR

Anyone have a patch of these "community" fixes?  If so I'll just fork the repo, apply the patch, and then we'll have a functioning Valheim+ with the **_code in public space_**.

**Comment by vividnightmare** [comment]
https://github.com/Grantapher/ValheimPlus

Wait someone already did exactly that :)

**Comment by Streamweaver** [comment]
Except that is temporary and now that one doesn't work with the new update either.

**Comment by bennysp** [comment]
It is updated now: https://github.com/Grantapher/ValheimPlus/releases/tag/0.9.9.15

Works on AMP+Docker dedicated server.
